### PR TITLE
Harvesters pagination

### PR DESCRIPTION
**Pull request by seeden**
added support for harvesters and plots pagination

### CODE REVIEW

File: packages/gui/src/components/plot/PlotHarvester.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  const { nodeId, host, expanded: expandedDefault = false } = props;
+
+  const { plots, noKeyFilenames, failedToOpenFilenames, duplicates, totalPlotSize, initialized } = useGetHarvesterQuery({
+    nodeId,
+  });
+
+  const [activeTab, setActiveTab] = useState<'PLOTS' | 'NOT_FOUND' | 'FAILED' | 'DUPLICATE'>('FAILED');
+  const [expanded, toggleExpand] = useToggle(expandedDefault);
+  const simpleNodeId = `${nodeId.substr(0, 6)}...${nodeId.substr(nodeId.length - 6)}`;
+  const isLocal = host === '127.0.0.1';
```
Comment (paninaro) [inline packages/gui/src/components/plot/PlotHarvester.tsx:32]: Do we need to handle IPv6 localhost as well? `::1`

File: packages/gui/src/components/plot/PlotHarvester.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+  useEffect(() => {
+    if ((activeTab === 'NOT_FOUND' && !noKeyFilenames) || (activeTab === 'FAILED' && !failedToOpenFilenames) || (activeTab === 'DUPLICATE' && !duplicates)) {
+      setActiveTab('PLOTS');
+    }
+  }, [activeTab, plots, noKeyFilenames, failedToOpenFilenames, duplicates]);
+
+  function hadneChangeActiveTab(newActiveTab) {
```
Comment (paninaro) [inline packages/gui/src/components/plot/PlotHarvester.tsx:40]:
```suggestion
  function handleChangeActiveTab(newActiveTab) {
```

File: packages/gui/src/components/plot/PlotHarvester.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                )}
+              </Typography>
+              <PlotHarvesterState nodeId={nodeId} />
+            </Flex>
+          </Flex>
+        </Flex>
+        <Flex alignItems="center">
+          <Tabs
+            value={activeTab}
+            onChange={(_event, newValue) => hadneChangeActiveTab(newValue)}
```
Comment (paninaro) [inline packages/gui/src/components/plot/PlotHarvester.tsx:81]:
```suggestion
            onChange={(_event, newValue) => handleChangeActiveTab(newValue)}
```

File: packages/gui/src/components/plot/PlotHarvesters.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import React, { useMemo } from 'react';
+import { orderBy } from 'lodash';
+import { useGetHarvestersSummaryQuery } from '@chia/api-react';
+import { Trans } from '@lingui/macro';
+import { Loading, Flex } from '@chia/core';
+import { Typography } from '@mui/material';
+import PlotHarvester from './PlotHarvester';
+
+function getIpAddress(harvester) {
+  if (harvester.connection.host === '127.0.0.1') {
```
Comment (paninaro) [inline packages/gui/src/components/plot/PlotHarvesters.tsx:10]: Same question re: IPv6 localhost `::1`

**Review by paninaro — APPROVED** [review]
Left a few comments/suggestions, but lgtm.

File: packages/gui/src/components/plot/PlotHarvester.tsx (hunk 1 above)
Comment (seeden) [reply packages/gui/src/components/plot/PlotHarvester.tsx:32]: I already added it. Thank you

File: packages/gui/src/components/plot/PlotHarvesters.tsx (hunk 4 above)
Comment (seeden) [reply packages/gui/src/components/plot/PlotHarvesters.tsx:10]: I already added it. Thank you

File: packages/gui/src/components/plot/PlotHarvester.tsx (hunk 3 above)
Comment (seeden) [reply packages/gui/src/components/plot/PlotHarvester.tsx:81]: Updated

**Review by paninaro — APPROVED** [review]

**Pull request merged**